embedded-kafka/src/test/java/com/playtika/test/kafka/camel/ProductionRouteTest.java

```diff
@@ -28,50 +28,51 @@
 import org.apache.camel.CamelContext;
 import org.apache.camel.ProducerTemplate;
+import org.apache.camel.ServiceStatus;
 import org.apache.camel.component.kafka.KafkaConstants;
-import org.apache.camel.test.spring.CamelSpringRunner;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
 
 import static com.playtika.test.kafka.properties.KafkaConfigurationProperties.KAFKA_BEAN_NAME;
 import static com.playtika.test.kafka.properties.ZookeeperConfigurationProperties.ZOOKEEPER_BEAN_NAME;
 import static java.util.Arrays.asList;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
-@SpringBootTest(
-        classes = {
-                ProductionRouteTest.TestConfiguration.class,
-                SampleProductionRouteContext.class
-        },
-        properties = "embedded.kafka.topicsToCreate=helloTopic"
-)
-@RunWith(CamelSpringRunner.class)
+@SpringBootTest
+@TestPropertySource(properties = "embedded.kafka.topicsToCreate=helloTopic")
+@ContextConfiguration(classes = {
+        ProductionRouteTest.TestConfiguration.class,
+        SampleProductionRouteContext.class
+})
+@EnableAutoConfiguration
 public class ProductionRouteTest {
 
-    @Autowired
-    ConfigurableListableBeanFactory beanFactory;
-    @Autowired
-    private CamelContext camelContext;
-    @Autowired
-    private ProducerTemplate producerTemplate;
-    @Autowired
-    private SampleRouteConfiguration routeConfiguration;
+    private final ConfigurableListableBeanFactory beanFactory;
+    private final CamelContext camelContext;
+    private final ProducerTemplate producerTemplate;
+    private final SampleRouteConfiguration routeConfiguration;
+    private final SampleTestRouteMonitor routeMonitor;
 
-    private SampleTestRouteMonitor routeMonitor;
-
-    @Before
-    public void setUp() throws Exception {
-        if (routeMonitor == null) {
-            routeMonitor = new SampleTestRouteMonitor(camelContext);
-        }
+    @Autowired
+    public ProductionRouteTest(
+            ConfigurableListableBeanFactory beanFactory,
+            CamelContext camelContext,
+            ProducerTemplate producerTemplate,
+            SampleRouteConfiguration routeConfiguration) throws Exception {
+        this.beanFactory = beanFactory;
```

Comment:
why not autowired as was before?

Reply:
It is autowired, but using constructor injection and not field injection. Why constructor injection ? There is many blog posts on the internet that favor constructor injection oven field injection. Here one of them https://stackoverflow.com/questions/19381846/why-use-constructor-over-setter-injection-in-cdi

Guys from Spring also strongly recommend constructor injection. 

But it is up to you. If you prefer field injection, I can easily put it back

Reply:
Constructor injection definitely makes sense, and, if we are talking about POJO services with logic, is way better than autowired  + componentScan
But here you switched for Test class, which is not a @Bean which you would pack into module and reuse)
So, for Test classes it is OK to @autowire stuff

Reply:
Done!